src/main/java/org/troyargonauts/subsystems/Elevator.java

```diff
@@ -0,0 +1,125 @@
+package org.troyargonauts.subsystems;
+import com.revrobotics.*;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.PIDCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.troyargonauts.Constants;
+import org.troyargonauts.Robot;
+
+/**
+ * Elevator Code: Extends elevator to max length, determined by limit switches.
+ * Elevator extends to score cone or cube which is taken in by intake
+ *
+ * @author TeoElRey, sgowda260
+ */
+public class Elevator extends SubsystemBase {
+    private final CANSparkMax leftMotor;
+    private final CANSparkMax rightMotor;
+    private SparkMaxLimitSwitch upperLimitSwitch;
+    private SparkMaxLimitSwitch lowerLimitSwitch;
+    // private boolean upperLimitSwitchValue;
+    // private boolean lowerLimitSwitchValue;
+    private PIDController pid;
+
+
+
+    /**
+     * Instantiates the motor controllers, limit switches, encoder, and PID controller for the Elevator. Left side motor is reversed
+     * Additionally, the left side motor is set to be inverted
+     * Encoders are built into the motor
+     * Motors are set to brake mode, meaning they will quickly stop when reaching limit instead of slowing on their own
+     * soft limit is set to 7, meaning motors will have a limit of 7 rotations backwards
+     */
+    public Elevator() {
+        leftMotor = new CANSparkMax(Constants.Elevator.LEFT, CANSparkMaxLowLevel.MotorType.kBrushless);
+        rightMotor = new CANSparkMax(Constants.Elevator.RIGHT, CANSparkMaxLowLevel.MotorType.kBrushless);
+
+        leftMotor.setInverted(false);
+        rightMotor.setInverted(false);
+
+        leftMotor.setIdleMode(CANSparkMax.IdleMode.kBrake);
+        rightMotor.setIdleMode(CANSparkMax.IdleMode.kBrake);
+
+        leftMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, 7);
+        rightMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, 7);
+
+//        upperLimitSwitch = leftMotor.getForwardLimitSwitch(SparkMaxLimitSwitch.Type.kNormallyClosed);
+//        upperLimitSwitch = rightMotor.getForwardLimitSwitch(SparkMaxLimitSwitch.Type.kNormallyClosed);
+//
+//        lowerLimitSwitch = leftMotor.getForwardLimitSwitch(SparkMaxLimitSwitch.Type.kNormallyClosed);
+//        lowerLimitSwitch = rightMotor.getForwardLimitSwitch(SparkMaxLimitSwitch.Type.kNormallyClosed);
+        pid = new PIDController(Constants.Elevator.kP, Constants.Elevator.kI ,Constants.Elevator.kD, Constants.Elevator.PERIOD);
+    }
+
+    /**
+     * Periodic will constantly check the encoder and limit switch values.
+     * These values will later be used to make sure elevator extends or retracts to optimal position.
+     */
+    @Override
+    public void periodic() {
```

Comment:
Set variables for encoder values and set their values here. Then use the variables in PIDs instead of getPosition().